src/staker.cairo

```diff
@@ -104,6 +122,8 @@ pub mod Staker {
         amount_delegated: Map<ContractAddress, u128>,
         delegated_cumulative_num_snapshots: Map<ContractAddress, u64>,
         delegated_cumulative_snapshot: Map<ContractAddress, Map<u64, DelegatedSnapshot>>,
+        
+        staking_log: Vec<StakingLogRecord>,
```

Comment:
is there a limit to the stored vector length? we use maps elsewhere, e.g. `delegated_cumulative_snapshot` and `delegated_cumulative_num_snapshots`

Reply:
As far as I understand the limit of Vec length in limited by 1 felt252 variable. which allows for 2**252, which, if we store record every millisecond will allow us to store records for `2294839414425501299513293321219508717341914344633675298312881700864` years.

Reply:
I see. I don't think that size is the problem. But yeah seems that using Maps consumes less gas. Will do a bit of a research. Thank you for pointing that out!